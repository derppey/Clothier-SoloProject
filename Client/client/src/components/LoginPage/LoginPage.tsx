import React from 'react'
import { useState } from 'react';
import { Link } from "react-router-dom";
import apiService from '../../apiServices';
import '../../styles/login.css';
import logo from '../../utils/ClothierWideRed.png'


const initialState = {
  email: '',
  password: '',
};



export default function LogIn(props: {setAuthenticated:Function}) : JSX.Element {
  const [userLogin, setUserLogin] = useState<{email:string, password:string}>(initialState);
  
  const handleEvent = (e:any)=> {
    const { name, value } = e.target;
    setUserLogin((prevState) => ({
      ...prevState,
      [name]: value,
    }));
  }

  const handleSubmit = async (e:React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const { email, password } = userLogin;
    const user = { email, password };
    const res = await apiService.login(user);
    
    if (res.error) {
      alert(`${res.message}`);
      setUserLogin(initialState);
    } else {
      const { accessToken } = res;
      localStorage.setItem('accessToken', accessToken);
      props.setAuthenticated(true);
    }
  }

  return (
    <div className='login-page is-flex is-justify-content-center'>
      <img className='logo' src={`${logo}`} alt="n/a"/>
      <div className='login-window box'>
        <h1 className='title is-3'>Welcome to Clothier!</ h1>
        <div className="field">
          <form onSubmit={handleSubmit}>
            <input className='mt-1 input' type='text' required name='email' placeholder="Email" value={userLogin.email} onChange={handleEvent}/>
            <input className='mt-5 input' type='password' required placeholder='Password' name='password' value={userLogin.password} onChange={handleEvent}/>
            <button className="mt-5 button is-primary is-rounded" type='submit'> Log In </button>
          </form>
          {/* <button className="button" type='click'> Log In with Facebook </button>
          <button className="button" type='click'> Log In with Google </button> */}
          <h1 className='mt-3 title is-5'>New to Clothier?</ h1>
          <Link to="/register">
            <button className="mt-0 button is-link is-rounded"> Register! </button>
          </Link>
        </div>
      </div>
    </div>
  )
}

