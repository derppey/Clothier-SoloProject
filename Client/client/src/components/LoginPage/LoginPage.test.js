import {render, screen} from '@testing-library/react';
import LoginPage from './LoginPage';

import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

jest.mock('../../apiServices.tsx',  () => ({
  login: () => ({email: 'test@test.com', password: 'test'})
}))

describe('Login Page', () => {
  test('should match the snapshot', () => {
    const { container } = render(<Router><LoginPage /></Router>);
    expect(container.firstChild).toMatchSnapshot();
  })

  test('should have correct title', () => {
    render(<Router><LoginPage /></Router>);
    screen.getByText(/Welcome to Clothier!/);
    screen.getByText(/New to Clothier?/);
  })
 

  test('Should call login with the correct credentials', async () => {
    // const setUser = jest.fn();
    const setAuthenticated = jest.fn();
    const setUserLogin = jest.fn();
    render(<Router><LoginPage setAuthenticated={setAuthenticated} /></Router>);
    const emailInput = screen.getByPlaceholderText(/email/i);
    const passwordInput = screen.getByPlaceholderText(/Password/i)
    const submitBtn = screen.getByRole('button', {name: /log in/i})

    userEvent.type(emailInput, 'test@test.com')
    userEvent.type(passwordInput, 'test')

    await userEvent.click(submitBtn)
    expect(setAuthenticated).toBeCalledTimes(1)
    expect(setUserLogin).toBeCalledTimes(0)

  })
})