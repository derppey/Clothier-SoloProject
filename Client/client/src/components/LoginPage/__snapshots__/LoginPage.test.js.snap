// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should match the snapshot 1`] = `
<div
  class="login-page is-flex is-justify-content-center"
>
  <img
    alt="n/a"
    class="logo"
    src="ClothierWideRed.png"
  />
  <div
    class="login-window box"
  >
    <h1
      class="title is-3"
    >
      Welcome to Clothier!
    </h1>
    <div
      class="field"
    >
      <form>
        <input
          class="mt-1 input"
          name="email"
          placeholder="Email"
          required=""
          type="text"
          value=""
        />
        <input
          class="mt-5 input"
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <button
          class="mt-5 button is-primary is-rounded"
          type="submit"
        >
           Log In 
        </button>
      </form>
      <h1
        class="mt-3 title is-5"
      >
        New to Clothier?
      </h1>
      <a
        href="/register"
      >
        <button
          class="mt-0 button is-link is-rounded"
        >
           Register! 
        </button>
      </a>
    </div>
  </div>
</div>
`;
